analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -0,0 +1,194 @@
+        }
+
+    /**
+     * Return a [Scope] created from the given type of [relation] for [projectPackage] in [spdxDocument], or `null` if
+     * there are no such relations.
+     */
+    private fun createScope(spdxDocument: SpdxDocument, projectPackage: SpdxPackage, relation: SpdxRelationship.Type) =
+        getDependencies(projectPackage, spdxDocument, relation).takeUnless { it.isEmpty() }?.let { dependencies ->
+            Scope(
+                name = relation.name.removeSuffix("_DEPENDENCY_OF").toLowerCase(),
+                dependencies = dependencies
+            )
+        }
+
+    override fun resolveDependencies(definitionFile: File): List<ProjectAnalyzerResult> {
+        val workingDir = definitionFile.parentFile
+        val spdxDocument = SpdxModelMapper.read(definitionFile, SpdxDocument::class.java)
+
+        val projectPackage = spdxDocument.packages.singleOrNull { it.packageFilename.isEmpty() }
+
+        val project = if (projectPackage != null) {
+            val scopes = listOfNotNull(
+                Scope(
+                    name = DEFAULT_SCOPE_NAME,
+                    dependencies = getDependencies(projectPackage, spdxDocument)
+                ),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.BUILD_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.DEV_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.OPTIONAL_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.PROVIDED_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.RUNTIME_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.TEST_DEPENDENCY_OF)
+            ).toSortedSet()
+
+            Project(
+                id = projectPackage.toIdentifier(),
+                definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                declaredLicenses = sortedSetOf(projectPackage.licenseDeclared),
+                vcs = VcsInfo.EMPTY,
+                vcsProcessed = processProjectVcs(workingDir, VcsInfo.EMPTY, projectPackage.homepage),
+                homepageUrl = projectPackage.homepage,
+                scopes = scopes
+            )
+        } else {
+            // TODO: How to deals with relationships between SPDX packages if we do not have a project to attach
+            //       dependencies to?
+            Project.EMPTY
+        }
+
+        val nonProjectPackages = if (projectPackage != null) {
+            spdxDocument.packages - projectPackage
+        } else {
+            spdxDocument.packages
+        }
+
+        val packages = nonProjectPackages.mapTo(sortedSetOf()) { pkg ->
+            val packageDescription = pkg.description.takeUnless { it.isEmpty() }
+                ?: pkg.summary.takeUnless { it.isEmpty() }
+                ?: pkg.comment
```

Comment:
`pkg.description` and `pkg.summary` are from the original packages meta data, while `pkg.comment` is from the author of the SPDX Document. How about removing the fallback #2?

Reply:
Done.